hudi-client/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -266,7 +256,7 @@ private void initialize(JavaSparkContext jsc, HoodieTableMetaClient datasetMetaC
   }
 
   private void initTableMetadata() {
-    this.metadata = new HoodieBackedTableMetadata(hadoopConf.get(), datasetWriteConfig.getBasePath(), datasetWriteConfig.getSpillableMapBasePath(),
```

Comment:
The reader-write pairs in metadata are related. So this can be simplified to be simply new HoodieBackedTableMetadata. 

HoodieBackedTableMetadata cannot work with any other implementaton of HoodieTableMetadata.